server/data/strapiApiTypes.ts

```diff
@@ -22,22 +22,31 @@ export type SingleResponse<T> = {
   meta?: Record<string, never>
 }
 
+type HasComponent = { component: SingleResponse<Component> }
 type HasComponents = { components: ListResponse<Component> }
+
+type HasProduct = { product: SingleResponse<Product> }
+type HasProducts = { products: ListResponse<Product> }
+type HasProductSet = { product_set: SingleResponse<ProductSet> }
 type HasTeam = { team: SingleResponse<Team> }
 type HasServiceArea = { service_area: SingleResponse<StrapiServiceArea> }
-type HasProduct = { product: SingleResponse<ProductWithTeamAndServiceArea> }
+type HasNamespace = { ns: SingleResponse<Namespace> }
+type HasEnvironments = { envs: ListResponse<Environment> }
 
-type ProductWithTeamAndServiceArea = Omit<Product, 'team' | 'serviceArea'> & HasTeam & HasServiceArea
+export type Product = Omit<components['schemas']['Product'], 'components' | 'team' | 'service_area' | 'product_set'> &
```

Comment:
These all remove the old field values and add the "correct" named types.. 

This is needed because strapi doesn't reuse named types and instead generates broken versions that are incomplete.. 

This adds all the possible relations to all possible types and makes it easier to consistently refer to the same thing